Userland/Libraries/LibJS/Parser.h

```diff
@@ -107,15 +108,31 @@ class Parser {
         {
             if (!position.has_value())
                 return {};
-            // We need to modify the source to match what the lexer considers one line - normalizing
-            // line terminators to \n is easier than splitting using all different LT characters.
-            String source_string { source };
-            source_string.replace("\r\n", "\n");
-            source_string.replace("\r", "\n");
-            source_string.replace(LINE_SEPARATOR, "\n");
-            source_string.replace(PARAGRAPH_SEPARATOR, "\n");
             StringBuilder builder;
-            builder.append(source_string.split_view('\n', true)[position.value().line - 1]);
+            GenericLexer lexer(source);
+            // Skip to the line we want
+            size_t current_line = 0;
+            dbgln_if(LEXER_DEBUG, "Generating source location hint for line: {}", position.value().line);
+            while (current_line < position.value().line - 1) {
+                if (lexer.consume_specific("\n") || lexer.consume_specific("\r\n") || lexer.consume_specific(LINE_SEPARATOR) || lexer.consume_specific(PARAGRAPH_SEPARATOR))
+                    current_line++;
+                else
+                    lexer.ignore();
+                if (lexer.is_eof()) {
+                    // Check if we are at the last line
+                    if (current_line == (position.value().line - 1))
+                        break;
+                    dbgln("Reached EOF during source hint generation; invalid input line: {}, only got to line {}", position.value().line, current_line);
```

Comment:
is this still happening with the second commit?

Reply:
The last line check restores the original behavior; in that if the hint should be 1 line past the last line it pretends like there is no line.
The second commit simply fixes a bug, in that the last byte would not be properly read if it was a line terminator. This bug only became apparent when using Windows-style line endings, as it would actually detect the last line terminator instead of ignoring it (both \r and \n are considered line terminators in the code, with the \n being ignored.)